### PR TITLE
[#175848172] Clean PM Client module and dependencies

### PR DESCRIPTION
**Pull request by GiovanniMancini**
**Decription**
This PR applies some cleanings to the PM Client Module and its dependencies. Other ongoing activities that will improve modularity are:
- Factor out `retryLogicForTransientResponseError` (different PR)
- Factor out (or get rid of) types needed to patch the client (depends on blocking activities)
- Refactor of fetches to rely on `io-ts-commons` or external packages in a browser environment (different story)

### CODE REVIEW

**Review by infantesimone — COMMENTED** [review]
Good work, but I have two considerations:
- integrations test in a dedicated `__integrations__ `folder (as suggested in IO-RFC-51);
- tests defined in `pagopaApi.test.ts` use a mock server, so they are integration tests? not unit tests?

File: src/types/pagopa.ts [hunk 1]
```diff
@@ -380,24 +377,10 @@ type Pay = t.TypeOf<typeof Pay>;
 export const PayRequest = repP(PayRequestPagoPA, 'data', Pay, 'PayRequest');
 
 export type PayRequest = t.TypeOf<typeof PayRequest>;
-
+// NEEDED
```
Comment (gunzip) [inline src/types/pagopa.ts:380]: what's the meaning of 'needed' in this context? (everything else is optional?)

Comment (GiovanniMancini) [reply src/types/pagopa.ts:380]: As you may have seen I tend to overcomment the code :-) but in this case is just a typo. I'll fix it

**Comment by GiovanniMancini** [comment]
@infantesimone I understand your position and I'll fix the naming scheme as you suggested, but even if we adopt a _narrow integration_ testing strategy as suggested by Martin Flower in this reference https://martinfowler.com/bliki/IntegrationTest.html there is a minimum of integration required to call a test _integration test_. In my opinion the bare minimum is

1. you test at least 2 components
2. the two component are independently unit tested, which is false in this case, since the fake server is not tested at all,
3. the two components are both component of the system, which is false too, since the server is not the real server.

As a surrogate, it could be acceptable to integrate a component with a _validated_ double of another component (as in the integration test with the PM test server), but in our case the fake server is not validated... I'm sure it also violates the _interface contract_, so it's useful only to make assertion on the client.
In my opinion, reusing a terminology developed in a context of object oriented development, such kind of tests are not integration tests but  _sociable unit tests_ as opposed to _solitary unit tests_ . If you are interested in a such useless :-)  distinction please check the following references
https://martinfowler.com/bliki/UnitTest.html
https://martinfowler.com/articles/mocksArentStubs.html

**Comment by infantesimone** [comment]
LGTM and @GiovanniMancini thanks for the explanation 💪

**Review by infantesimone — APPROVED** [review]

File: src/api/pagopa.ts [hunk 2]
```diff
@@ -175,7 +174,7 @@ const getWallets: GetWalletsUsingGETExtraT = {
   headers: ParamAuthorizationBearerHeader,
   response_decoder: getPatchedWalletsUsingGETDecoder(WalletListResponse),
 };
-
+/*
```
Comment (cloudify) [inline src/api/pagopa.ts:177]: don't commit commented out code

**Review by cloudify — APPROVED** [review]

**Pull request merged**